src/metadata.rs

```diff
@@ -0,0 +1,41 @@
+use libc::{c_char, c_void, c_int};
+use utils::{_string};
+use std::ffi::{CString};
+use gdal_major_object::MajorObject;
+
+#[link(name="gdal")]
+extern {
+    fn GDALGetMetadataItem(hGdalMayorObject: *const c_void, pszName: *const c_char, pszDomain: *const c_char) -> *const c_char;
+    fn GDALSetMetadataItem(hGdalMayorObject: *const c_void, pszName: *const c_char, pszValue: *const c_char, pszDomain: *const c_char ) -> c_int;
+}
+
+pub trait Metadata: MajorObject {
+
+    fn get_metadata_item(&self, key: &str, domain: &str) -> Option<String> {
+        let c_key = CString::new(key.to_owned()).expect("Could not transform String to CString"); // FIXME: no unwrap
+        let c_domain = CString::new(domain.to_owned()).expect("Could not transform String to CString");
```

Comment:
Need any help making this changes? I can make them for you if you want, just let me know